packages/client/components/SendCommentButton.tsx

```diff
@@ -41,29 +42,52 @@ interface Props {
   commentSubmitState: CommentSubmitState
   onSubmit: () => void
   dataCy: string
+  isSubmitDisabled: boolean
 }
 
 const SendCommentButton = (props: Props) => {
-  const {commentSubmitState, onSubmit, dataCy} = props
+  const {commentSubmitState, onSubmit, dataCy, isSubmitDisabled} = props
   const {
     tooltipPortal,
     openTooltip,
     closeTooltip,
     originRef: tipRef
   } = useTooltip<HTMLButtonElement>(MenuPosition.LOWER_CENTER)
-  const isDisabled = commentSubmitState === 'idle'
+
+  const androidCommentSubmitState = isSubmitDisabled ? 'typing' : 'idle'
+
+  useEffect(() => {
+    const btn = document.querySelector('.send-comment-button')
+    const submitListener = () => {
+      if (isAndroid) {
+        btn?.addEventListener('touchend', (e) => {
+          e.preventDefault()
+          onSubmit()
+        })
+      } else {
+        onSubmit()
+      }
+    }
+    btn?.addEventListener('click', submitListener, true)
+    return () => {
+      btn?.removeEventListener('click', submitListener, true)
+    }
+  }, [onSubmit])
```

Comment:
-2 there's no need for `useEffect` here. what's more, when on android you're adding `touchend` event listener which isn't cleared and will cause a memory leak. In that kind of situation, it's worth asking why is this button disabled, when the text is already there? 

The root cause seems to be the function, that checks if the editor has text. On android we use fallback editor, and this function doesn't take that into account. see the diff below:

<img width="1124" alt="Screenshot 2022-11-04 at 07 29 30" src="https://user-images.githubusercontent.com/1017620/199906563-cb24e32a-33d6-4a45-9692-243c6369fe7d.png">

changing the way we get the current text, fixes the issue with only a few lines of code without a need for that kind of `useEffects`.

Reply:
there's a good read from react team about when you _don't_ need `useEffect`: https://beta.reactjs.org/learn/you-might-not-need-an-effect

Reply:
@BartoszJarocki Thank you for your feedback and suggestions; the other changes were made to account for the remaining acceptance criteria; please take another look.